pycqed/analysis_v3/processing_pipeline.py

```diff
@@ -1,353 +1,774 @@
+
+From here on I will refer to the processing functions in the pipeline as
+nodes.
+
+Instructions for use:
+    Initialization
+        - without any input arguments: ProcessingPipeline()
+        - or with input parameters:
+         ProcessingPipeline(node_name, **node_params), where node_name is
+         the name of the node, and **node_params all the parameters
+         required by the node including the necessary keys described above
+        ! For ease of use, keys_in can also be specified as
+            - 'raw': the raw data corresponding to the measured object
+            - 'previous': the keys_out of the previous node dictionary
+             for the measured object.
+            - 'previous measured_object_name.node_name': the keys_out of the
+             dictionary for the measured object which has the node_name
+        ! keys_out do not need to be specified by the user as they will be
+         automatically constructed from the measured object name and the
+         keys_in
+
+            Examples:
+                ProcessingPipeline('average_data',
+                                    keys_in='raw',
+                                    shape=(3,2),
+                                    meas_obj_names='TWPA')
+                ProcessingPipeline('ramsey_analysis',
+                                    keys_in='previous rotate_iq',
+                                    meas_obj_names=['qb1', 'qb2'])
+
+    Adding processing node dictionaries:
+        - to add more node dictionaries to the pipeline, call the "add_node"
+         method with the same "node_name" and **node_params arguments as
+         described above under "Initialization."
+         
+            Example: same as above but replace ProcessingPipeline with 
+            ProcessingPipeline_instance.add_node
+    
+    Up to now, the pipeline is just a list of dictionaries with the
+    key-value pairs as provided by the user:
+        
+        Example of a "raw" pipeline: 
+            [{'keys_in': 'raw',
+              'shape': (80, 10),
+              'meas_obj_names': ['qb2'],
+              'node_name': 'average_data'},
+             {'keys_in': 'previous qb2.average_data',
+              'shape': (10, 8),
+              'averaging_axis': 0,
+              'meas_obj_names': ['qb2'],
+              'update_key': False,
+              'node_name': 'average_data'},
+             {'meas_obj_names': ['qb2'],
+              'keys_out': None,
+              'keys_in': 'previous qb2.average_data1',
+              'std_keys': 'previous qb2.get_std_deviation1',
+              'node_name': 'SingleQubitRBAnalysis'}]
+    
+    Creating the pipeline:
+        - the analysis framework always expects keys_in to be a list of 
+         keys in the data_dict, and most functions expect keys_out
+        - to create the pipeline that will be used by the analysis 
+         framework, the user must call: 
+         ProcessingPipeline_instance(meas_obj_value_names_map), where 
```

Comment:
this documentation might be outdated (seems that meas_obj_value_names_map is no longer passed at init stage)

Reply:
It is not completely outdated. The pipeline instance must still be called with meas_obj_value_names map in order to resolve it. However, now I added the option to pass the un-resolved pipeline to the analysis and the analysis resolves it based on meas_obj_value_names_map. So the user does not necessarily need to do it, and I've updated this part of the docstring to say so.